### DISCUSSION TITLE
[Question]: How can I define pi and reflecting my pi to sin, cos, tan?

### DISCUSSION THREAD

**Discussion by hagozaebii** [cvc5/cvc5#11052] [Q&A] [answered]
Hello, I am a person interested in how pi functions in CVC5. 

Specifically, I would like to know if there is a way to set appropriate bounds for the real.pi value and then have functions like sin, cos, and tan operate on pi within these bounds that I have defined.
Thank you for taking the time to read this.

**Comment by hagozaebii** [comment]
Also, I can see non-standard theories(and non-standard functions) here(https://github.com/cvc5/cvc5/blob/88dc94ea1861377c7f1aa16dd21f3ba797edd902/docs/theories/theories.rst)
I guess that the page does not involve some non-standard functions. For instance, `iand` or `constant array`. The functions are standard? Or, these are just omitted?

**Comment by ajreynol** [comment, marked as answer]
Sorry for the delay in reply.  The function `real.pi` is available in the parser and our API (Kind::PI) and is given bounds internally.  You can refine these bounds yourself in the input if you want e.g. `(assert (<= lower real.pi upper))`, which could help cvc5 solve the problem, as the transcendental solver in cvc5 uses this symbol when solving problems with transcendentals.
